06.bowling_object/game.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+require_relative 'shot'
+
+class Game
+  LAST_FRAME = 10
+
+  def initialize(marks)
+    @frames = []
+    frame_marks = []
+    frame_idx = 0
+
+    marks.each do |mark|
+      frame_marks << mark
+      if frame_idx + 1 != LAST_FRAME
+        if frame_marks[1] || frame_marks.first == 'X'
+          @frames << Frame.new(frame_marks)
+          frame_marks = []
+          frame_idx += 1
+        end
+      else
+        @frames << Frame.new(frame_marks)
+        frame_marks = []
+      end
+    end
+  end
+
+  def score
+    game_score = 0
+    frame_idx = 0
+
+    @frames.each_with_index do |frame, index|
+      game_score += frame.score
+      next if frame_idx + 1 == LAST_FRAME
+
+      if frame.strike?
+        game_score += strike_bonus(index)
+      elsif frame.spare?
+        game_score += spare_bonus(index)
+      end
+      frame_idx += 1
+    end
+    game_score
```

Comment:
```suggestion
    @frames.sum do |frame|
      frame.score(@frames)
    end
```

こんな感じで他のフレームの情報を引数で渡すことで、点数の計算は全部frame自身で行う、というロジックが作れるかも。
フレーム番号はFrame.newするタイミングで渡すと良さそうです

Reply:
`sum`メソッドを使用したループ処理は私にとって画期的でした！勉強になります！
`Frame`クラスで計算処理の役割を持たせることにより、Gameクラスの責務を減らすことができました。